Fix round-off error in hardware division

MJoergen opened the pull request:
This fixes issue #786

The problem was lack of precision in the intermediate calculations. This fix increases the used number of DSPs from 12 to 32.

MJoergen committed:
Fix round-off error in hardware division

This fixes issue #786

The problem was lack of precision in the intermediate calculations. This
fix increases the used number of DSPs from 12 to 32.

lydon42 reviewed (approved):
Looks good to me. Test build succeeds, Mandelbrot Explorer 65 still produces an Appleman.

MJoergen committed:
hardware_divider: Rewrite unit test

First, the expected value is now calculated automatically.
Second, the test is much easier to extend with new values.

MJoergen committed:
hardware_divider: Add more test cases
